src/components/CountryPerformanceOnRisk.js

```diff
@@ -46,36 +44,66 @@ export class CountryPerformanceOnRisk extends Component {
 
 
   updateValue(idxOfSelector, selectedCountry) {
+
+    let cloneSelectorConfig = this.state.selectorConfig.slice()
+    cloneSelectorConfig[idxOfSelector].country = undefined
     if (selectedCountry) {
-      this.props.dispatch({type: "addRemoveLine", id: selectedCountry.value + '1', idx: idxOfSelector})
-    } else {
-      this.props.dispatch({type: "addRemoveLine", id: selectedCountry, idx: idxOfSelector})
+      cloneSelectorConfig[idxOfSelector].country = selectedCountry.value
     }
 
     this.setState({
-      countrySelectors: Object.assign(
+      selectorConfig: Object.assign(
         [],
-        this.state.countrySelectors,
-        this.state.countrySelectors.slice()[idxOfSelector].selected  = selectedCountry
+        this.state.selectorConfig,
+        cloneSelectorConfig
       )
     })
   }
 
 
+  makePlotlyData(countryID, riskID, cubeByRiskByCountry) {
+    var dataTable = cubeByRiskByCountry[riskID][countryID];
+    return {
+      x: dataTable.map(row => row.date),
+      y: dataTable.map(row => row.count),
+      name: this.state.countries[countryID].title,
+      type: 'scatter',
+    }
+  }
+
+
   render() {
     let self = this
+    let plotlyData = []
+    let countries = []
+    if (this.state.cubeByRiskByCountry) {
```

Comment:
The overall logic of this method is quite complex. A small comment above it about what it does would be useful.